send-transaction-service/src/send_transaction_service.rs

```diff
@@ -804,12 +806,36 @@ mod test {
             &connection_cache,
             1000,
             1,
+            Arc::new(AtomicBool::new(false)),
         );
 
         drop(sender);
         send_transaction_service.join().unwrap();
     }
 
+    #[test]
+    fn validator_exit() {
+        let tpu_address = "127.0.0.1:0".parse().unwrap();
+        let bank = Bank::default_for_tests();
+        let bank_forks = Arc::new(RwLock::new(BankForks::new(bank)));
+        let (_sender, receiver) = unbounded();
+
+        let exit = Arc::new(AtomicBool::new(false));
+        let connection_cache = Arc::new(ConnectionCache::default());
+        let _send_transaction_service = SendTransactionService::new::<NullTpuInfo>(
+            tpu_address,
+            &bank_forks,
+            None,
+            receiver,
+            &connection_cache,
+            1000,
+            1,
+            exit.clone(),
+        );
+
+        exit.store(true, Ordering::Relaxed);
```

Comment:
i think exit is going to race the thread spawn here, should probably wait on... something.

similarly need to sit on `join()` after setting exit to ensure the thread stops as intended.

Reply:
> should probably wait on... something.

i guess easiest would be to switch to a zero-capacity bounded channel and do a blocking write to it. once thread is up, that message will be consumed and waiter will unblock. at that point it's safe to signal exit

Reply:
Yep, great idea. Done.